### ISSUE TITLE
c-s read command failed on start of destroy_data_then_repair nemesis

### ISSUE THREAD

**Issue by enaydanov** [scylladb/scylladb#17782]
## Packages

Scylla version: `5.5.0~dev-20240307.19bcea6216f4` with build-id `85250840bdc793b2a30610e4f3b9f0381b870d85`





Kernel Version: `5.15.0-1058-azure`


## Issue description

- [ ] This issue is a regression.
- [x] It is unknown if this issue is a regression.

Previous run passed but target nodes for nemeses were not the same, so, it's not clear if it's a regression or not.

DestroyDataThenRepair nemesis stopped a target node (10.0.0.5) as a first step. Despite we had another 3 nodes c-s read command failed to get quorum. Also, at the same moment c-s read command got authentication error for another node (10.0.0.6):

```
...
total,     170541092,    2875,    2875,    2875,     3.5,     3.3,     6.1,    10.4,    15.2,    27.2,71970.0,  0.00195,      0,      0,       0,       0,       0,       0                                         
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
com.datastax.driver.core.exceptions.TransportException: [longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042] Connection has been closed                                               
ERROR 03:18:18,798 Authentication error on host /10.0.0.6:9042: authentication failed                                                                                                                               
total,     170549585,    1699,    1699,    1699,     5.9,     3.7,    13.9,    16.7,    26.1,    69.1,71975.0,  0.00195,      0,      0,       0,       0,       0,       0                                         
WARN  03:18:19,764 Error creating netty channel to longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042                                                                                 
com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042                                   
Caused by: java.net.ConnectException: Connection refused                                                                                                                                                            
        at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)                                                                                                                                       
        at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)                                                                                                                         
        at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)                                                                                                 
        at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)                                                                                    
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)                                                                                                             
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)                                                                                                   
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)                                                                                                            
total,     170556998,    1483,    1483,    1483,     6.7,     4.1,    14.3,    18.2,    34.9,    51.6,71980.0,  0.00195,      0,      0,       0,       0,       0,       0                                         
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)                                                                                                                            
java.io.IOException: Operation x10 on key(s) [334c3250325032303630]: Error executing: (ReadFailureException): Cassandra failure during read query at consistency QUORUM (3 responses were required but only 2 replic
        at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)                                                                                            
                                                                                                                                                                                                                    
        at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)                                                                                                               
        at org.apache.cassandra.stress.Operation.error(Operation.java:141)                                                                                                                                          
        at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)                                                                                                   
        at org.apache.cassandra.stress.Operation.timeWithRetry(Operation.java:119)                                                                                                                                  
        at java.base/java.lang.Thread.run(Thread.java:829)                                                                                                                                                          
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:101)                                                                                                                
WARN  03:18:21,770 Error creating netty channel to longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042                                                                                 
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:109)                                                                                                                
com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042                                   
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:264)                                                                                                                
Caused by: java.net.ConnectException: Connection refused                                                                                                                                                            
        at org.apache.cassandra.stress.StressAction$Consumer.run(StressAction.java:473)                                                                                                                             
        at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)                                                                                                                                       
java.io.IOException: Operation x10 on key(s) [37374c36393436363531]: Error executing: (ReadFailureException): Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replic
        at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)                                                                                                                         
                                                                                                                                                                                                                    
        at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)                                                                                                 
        at org.apache.cassandra.stress.Operation.error(Operation.java:141)                                                                                                                                          
        at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)                                                                                    
        at org.apache.cassandra.stress.Operation.timeWithRetry(Operation.java:119)                                                                                                                                  
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)                                                                                                             
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:101)                                                                                                                
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)                                                                                                   
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:109)                                                                                                                
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)                                                                                                            
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:264)                                                                                                                
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)                                                                                                                            
        at org.apache.cassandra.stress.StressAction$Consumer.run(StressAction.java:473)                                                                                                                             
        at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)                                                                                            
total,     170557106,    1418,    1391,    1391,     6.9,     4.8,    15.0,    18.7,    19.7,    19.7,71980.1,  0.00195,      2,      0,       0,       0,       0,       0                                         
        at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)                                                                                                               
java.io.IOException: Operation x10 on key(s) [3235384d4f4d4d303131]: Error executing: (ReadFailureException): Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replic
        at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)                                                                                                   
                                                                                                                                                                                                                    
        at java.base/java.lang.Thread.run(Thread.java:829)                                                                                                                                                          
        at org.apache.cassandra.stress.Operation.error(Operation.java:141)                                                                                                                                          
WARN  03:18:25,772 Error creating netty channel to longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042                                                                                 
        at org.apache.cassandra.stress.Operation.timeWithRetry(Operation.java:119)                                                                                                                                  
com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042                                   
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:101)                                                                                                                
Caused by: java.net.ConnectException: Connection refused                                                                                                                                                            
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:109)                                                                                                                
        at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)                                                                                                                                       
        at org.apache.cassandra.stress.operations.predefined.CqlOperation.run(CqlOperation.java:264)                                                                                                                
        at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)                                                                                                                         
        at org.apache.cassandra.stress.StressAction$Consumer.run(StressAction.java:473)                                                                                                                             
        at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)                                                                                                 
        at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)                                                                                    
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)                                                                                                             
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)                                                                                                   
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)                                                                                                            
        at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)                                                                                                                            
        at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)                                                                                            
        at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)                                                                                                               
        at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)                                                                                                   
        at java.base/java.lang.Thread.run(Thread.java:829)                                                                                                                                                          
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 0 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 0 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 0 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 0 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replica responded, 1 failed)                              
com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 0 replica responded, 1 failed)                              
...
```

The only suspicious thing I see before this event on 10.0.0.6 is 
```
...
Mar 10 03:16:58.381817 longevity-tls-1tb-7d-master-db-node-eastus-2 scylla[6084]:  [shard  2:stmt] lsa-timing - compact took 40000 us, trying to release 0.250 MiB non-preemptibly, reserve: {goal: 0, max: 30}, at 
0x62c194e 0x62c1f60 0x62c2248 0x20f041f 0x20f69ba 0x20fe0e0 0x22fca89 0x5dbdf9f 0x5dbf287 0x5de3000 0x5d7e79a /opt/scylladb/libreloc/libc.so.6+0x8c946 /opt/scylladb/libreloc/libc.so.6+0x11296f                    
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<std::unique_ptr<cached_file::cached_page, cached_file::cached_p
age::cached_page_del> >, cached_file::get_page_ptr(unsigned long, unsigned long, tracing::trace_state_ptr)::{lambda(seastar::temporary_buffer<char>&&)#1}, seastar::future<seastar::temporary_buffer<char> >::then_i
mpl_nrvo<cached_file::get_page_ptr(unsigned long, unsigned long, tracing::trace_state_ptr)::{lambda(seastar::temporary_buffer<char>&&)#1}, seastar::future<std::unique_ptr<cached_file::cached_page, cached_file::ca
ched_page::cached_page_del> > >(cached_file::get_page_ptr(unsigned long, unsigned long, tracing::trace_state_ptr)::{lambda(seastar::temporary_buffer<char>&&)#1}&&)::{lambda(seastar::internal::promise_base_with_ty
pe<std::unique_ptr<cached_file::cached_page, cached_file::cached_page::cached_page_del> >&&, cached_file::get_page_ptr(unsigned long, unsigned long, tracing::trace_state_ptr)::{lambda(seastar::temporary_buffer<ch
ar>&&)#1}&, seastar::future_state<seastar::temporary_buffer<char> >&&)#1}, seastar::temporary_buffer<char> >                                                                                                        
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >, cached_file::get_page(unsign
ed long, unsigned long, tracing::trace_state_ptr)::{lambda(std::unique_ptr<cached_file::cached_page, cached_file::cached_page::cached_page_del>)#1}, seastar::future<std::unique_ptr<cached_file::cached_page, cache
d_file::cached_page::cached_page_del> >::then_impl_nrvo<cached_file::get_page(unsigned long, unsigned long, tracing::trace_state_ptr)::{lambda(std::unique_ptr<cached_file::cached_page, cached_file::cached_page::c
ached_page_del>)#1}, seastar::future<seastar::temporary_buffer<char> > >(cached_file::get_page(unsigned long, unsigned long, tracing::trace_state_ptr)::{lambda(std::unique_ptr<cached_file::cached_page, cached_fil
e::cached_page::cached_page_del>)#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >&&, cached_file::get_page(unsigned long, unsigned long, tracing::trace_state_ptr)::{lamb
da(std::unique_ptr<cached_file::cached_page, cached_file::cached_page::cached_page_del>)#1}&, seastar::future_state<std::unique_ptr<cached_file::cached_page, cached_file::cached_page::cached_page_del> >&&)#1}, st
d::unique_ptr<cached_file::cached_page, cached_file::cached_page::cached_page_del> >                                                                                                                                
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >, cached_file::stream::next():
:{lambda(seastar::temporary_buffer<char>)#1}, seastar::future<seastar::temporary_buffer<char> >::then_impl_nrvo<cached_file::stream::next()::{lambda(seastar::temporary_buffer<char>)#1}, seastar::future<seastar::t
emporary_buffer<char> > >(cached_file::stream::next()::{lambda(seastar::temporary_buffer<char>)#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >&&, cached_file::stream::n
ext()::{lambda(seastar::temporary_buffer<char>)#1}&, seastar::future_state<seastar::temporary_buffer<char> >&&)#1}, seastar::temporary_buffer<char> >                                                               
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<seastar::bool_class<seastar::stop_iteration_tag> >, cached_file
_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long
&, seastar::temporary_buffer<unsigned char>&) const::{lambda()#1}::operator()() const::{lambda(seastar::temporary_buffer<char>)#1}, seastar::future<seastar::temporary_buffer<char> >::then_impl_nrvo<cached_file_im
pl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long&, 
seastar::temporary_buffer<unsigned char>&) const::{lambda()#1}::operator()() const::{lambda(seastar::temporary_buffer<char>)#1}, seastar::future<seastar::bool_class<seastar::stop_iteration_tag> > >(cached_file_im
pl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long&, 
seastar::temporary_buffer<unsigned char>&) const::{lambda()#1}::operator()() const::{lambda(seastar::temporary_buffer<char>)#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::bool_class<seastar::s
top_iteration_tag> >&&, cached_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(
cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&) const::{lambda()#1}::operator()() const::{lambda(seastar::temporary_buffer<char>)#1}&, seastar::future_state<seastar::temporary_buf
fer<char> >&&)#1}, seastar::temporary_buffer<char> >                                                                                                                                                                
                                                                                     --------                                                                                                                       
                                                                                     seastar::internal::repeater<cached_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_
file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&) const::{lambda()#1}>                      
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<unsigned char> >, cached_file_impl::d
ma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long&, seast
ar::temporary_buffer<unsigned char>&) const::{lambda()#2}, seastar::future<void>::then_impl_nrvo<cached_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, u
nsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&) const::{lambda()#2}, seastar::future<seastar::temporary_bu
ffer<unsigned char> > >(cached_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(
cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&) const::{lambda()#2}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::temporary_buffer<unsigned char> >&&, cached_file
_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(cached_file::stream&, unsigned long&, seastar::temporary_buffer<unsigned char>&)#1}::operator()(cached_file::stream&, unsigned long
&, seastar::temporary_buffer<unsigned char>&) const::{lambda()#2}&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>                                                                                
                                                                                     --------                                                                                                                       

                                                                                     seastar::internal::do_with_state<std::tuple<cached_file::stream, unsigned long, seastar::temporary_buffer<unsigned char> >, sea
star::future<seastar::temporary_buffer<unsigned char> > >                                                                                                                                                           
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<unsigned char> >, tracking_file_impl:
:dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(reader_permit::resource_units)#1}::operator()(reader_permit::resource_units) const::{lambda(seastar::temporary_buffer<unsigned char>)#1},
 seastar::future<seastar::temporary_buffer<unsigned char> >::then_impl_nrvo<tracking_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(reader_permit::resource_units)#1}::operato
r()(reader_permit::resource_units) const::{lambda(seastar::temporary_buffer<unsigned char>)#1}, seastar::future<seastar::temporary_buffer<unsigned char> > >(tracking_file_impl::dma_read_bulk(unsigned long, unsign
ed long, seastar::io_intent*)::{lambda(reader_permit::resource_units)#1}::operator()(reader_permit::resource_units) const::{lambda(seastar::temporary_buffer<unsigned char>)#1}&&)::{lambda(seastar::internal::promi
se_base_with_type<seastar::temporary_buffer<unsigned char> >&&, tracking_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_intent*)::{lambda(reader_permit::resource_units)#1}::operator()(reader_p
ermit::resource_units) const::{lambda(seastar::temporary_buffer<unsigned char>)#1}&, seastar::future_state<seastar::temporary_buffer<unsigned char> >&&)#1}, seastar::temporary_buffer<unsigned char> >             
                                                                                     --------                                                                                                                       
                                                                                     seastar::continuation<seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >, seastar::file_data_source_im
pl::issue_read_aheads(unsigned int)::{lambda(seastar::future<seastar::temporary_buffer<unsigned char> >)#1}, seastar::future<seastar::temporary_buffer<unsigned char> >::then_wrapped_nrvo<seastar::future<seastar::
temporary_buffer<char> >, seastar::file_data_source_impl::issue_read_aheads(unsigned int)::{lambda(seastar::future<seastar::temporary_buffer<unsigned char> >)#1}>(seastar::file_data_source_impl::issue_read_aheads
(unsigned int)::{lambda(seastar::future<seastar::temporary_buffer<unsigned char> >)#1}&&)::{lambda(seastar::internal::promise_base_with_type<seastar::temporary_buffer<char> >&&, seastar::file_data_source_impl::is
sue_read_aheads(unsigned int)::{lambda(seastar::future<seastar::temporary_buffer<unsigned char> >)#1}&, seastar::future_state<seastar::temporary_buffer<unsigned char> >&&)#1}, seastar::temporary_buffer<unsigned c
har> >                                                                                                                                                                                                              
Mar 10 03:16:58.381832 longevity-tls-1tb-7d-master-db-node-eastus-2 scylla[6084]:  [shard  2:stmt] lsa-timing - - segments to release: 30                                                                           
Mar 10 03:16:58.381836 longevity-tls-1tb-7d-master-db-node-eastus-2 scylla[6084]:  [shard  2:stmt] lsa-timing - - processed 3 regions: reclaimed from 1, compacted 2                                                
Mar 10 03:16:58.381840 longevity-tls-1tb-7d-master-db-node-eastus-2 scylla[6084]:  [shard  2:stmt] lsa-timing - - evicted memory: 0.146 MiB                                                                         
Mar 10 03:16:58.381843 longevity-tls-1tb-7d-master-db-node-eastus-2 scylla[6084]:  [shard  2:stmt] lsa-timing - - compacted segments: 4                                                                             
Mar 10 03:16:58.381846 longevity-tls-1tb-7d-master-db-node-eastus-2 scylla[6084]:  [shard  2:stmt] lsa-timing - - compacted memory: 0.269 MiB                                                                       
...
```

## Impact

c-s read command failed to get quorum

## How frequently does it reproduce?

Looks like intermittent problem, maybe caused by environment.

## Installation details

Cluster size: 4 nodes (Standard_L16s_v3)

Scylla Nodes used in this run:
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-7 (57.151.106.21 | 10.0.0.14) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-6 (172.206.225.71 | 10.0.0.14) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-5 (20.121.141.232 | 10.0.0.14) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-4 (20.62.199.180 | 10.0.0.8) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-3 (20.62.198.161 | 10.0.0.7) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-2 (20.62.197.186 | 10.0.0.6) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-a6095eee-eastus-1 (20.62.199.49 | 10.0.0.5) (shards: 14)


OS / Image: `/subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/SCYLLA-IMAGES/providers/Microsoft.Compute/images/scylla-5.5.0-dev-x86_64-2024-03-08T02-41-32` (azure: undefined_region)

Test: `longevity-1tb-5days-azure-test`
Test id: `a6095eee-a478-42f1-8d4c-5b35b06ac790`
Test name: `scylla-master/tier1/longevity-1tb-5days-azure-test`
Test config file(s):

- [longevity-1TB-5days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/516aa4d9d6035d122255c3013e2b2101e1e3b9cb/test-cases/longevity/longevity-1TB-5days-authorization-and-tls-ssl.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor a6095eee-a478-42f1-8d4c-5b35b06ac790`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=a6095eee-a478-42f1-8d4c-5b35b06ac790)
- Show all stored logs command: `$ hydra investigate show-logs a6095eee-a478-42f1-8d4c-5b35b06ac790`


## Logs:
- **db-cluster-a6095eee.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/db-cluster-a6095eee.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/db-cluster-a6095eee.tar.gz)
- **sct-runner-events-a6095eee.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/sct-runner-events-a6095eee.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/sct-runner-events-a6095eee.tar.gz)
- **sct-a6095eee.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/sct-a6095eee.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/sct-a6095eee.log.tar.gz)
- **loader-set-a6095eee.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/loader-set-a6095eee.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/loader-set-a6095eee.tar.gz)
- **monitor-set-a6095eee.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/monitor-set-a6095eee.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/a6095eee-a478-42f1-8d4c-5b35b06ac790/20240310_033041/monitor-set-a6095eee.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/tier1/job/longevity-1tb-5days-azure-test/4/)
[Argus](https://argus.scylladb.com/test/1e333df8-a7e8-4171-8ab7-1d7bdea907d5/runs?additionalRuns[]=a6095eee-a478-42f1-8d4c-5b35b06ac790)
</details>

**Comment by mykaul** [comment]
```
Connection refused: longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042
```

Is very strange. Did 10.0.0.5 crash or something?

**Comment by michoecho** [comment]
> ```
> Connection refused: longevity-tls-1tb-7d-master-db-node-eastus-1.internal.cloudapp.net/10.0.0.5:9042
> ```
> 
> Is very strange. Did 10.0.0.5 crash or something?

It was shut down. This nemesis shuts down a node, removes some data, starts the node again and runs repair.

The suspicious part are the QUORUM reads which require 3 replies. This can't (or at least shouldn't) be coming from the regular reads. The test keyspace has rf=3, so QUORUM should be 2, and there aren't any topology operations in progress which could bump it up to 3.

But I guess a raft read on a 4-node cluster should require 3 replies. 

Given the above, and that the errors talk about an authentication failure, and that this test failure happened a day after auth-on-raft (#16578) was merged, I think this might have something to do with the auth-on-raft change.

**Comment by mykaul** [comment]
@nuivall - please see above - thoughts?

**Comment by nuivall** [comment]
Was the cluster started with consistent-topology-changes feature?

**Comment by nuivall** [comment]
If it isn't then likely duplicate of https://github.com/scylladb/scylladb/issues/15221 famility.

**Comment by michoecho** [comment]
> Was the cluster started with consistent-topology-changes feature?

No. It was started without it.

> If it isn't then likely duplicate of https://github.com/scylladb/scylladb/issues/15221 family.

Doesn't seem so. SCT sets the RF for `system_auth` appropriately, and if Scylla's logs are to be believed, `system_auth` should be replicated on all 4 nodes at the point of failure. And yet, when one of these nodes is stopped, the Scylla complains that it can't satisfy a QUORUM of 3, because only 2 replicas are alive. Something is fishy here.

**Comment by mykaul** [comment]
> > Was the cluster started with consistent-topology-changes feature?
> 
> No. It was started without it.

Ah, this is worth mentioning the initial bug description.

Regretfully, we need to support opt-out mode of topology over Raft, so this is a supported scenario.

CC @tzach

**Comment by nuivall** [comment]
Would be better to reproduce with raft, that's why we have the whole effort to fix problems with metadata consistency. Then if it reproduces we know there is something to fix. We can't fix in non-raft mode if raft is the fix...

Also btw I see it's on azure. Is it some new effort to test on azure? Maybe it's related to something totally different. I see only 4 (!) runs on argus.

**Comment by nuivall** [comment]
> > Would be better to reproduce with raft, that's why we have the whole effort to fix problems with metadata consistency (and avaliability in auth case) Then if it reproduces we know there is something to fix. We can't fix in non-raft mode if raft is the fix...
> 
> Unfortunately, we should still support non topology-over-raft (as opt-out) in 6.0. We could say that this bug is only fixed with it - that's legit.
> 
> > Also btw I see it's on azure. Is it some new effort to test on azure? Maybe it's related to something totally different. I see only 4 (!) runs on argus.
> 
> No, it isn't. We run some on Azure.

Then why we only have 4 runs stored? We can't do any regression reasoning based on such number.

**Comment by mykaul** [comment]
> Then why we only have 4 runs stored? We can't do any regression reasoning based on such number.

Perhaps just on master branch. We have many more across releases.

**Comment by ptrsmrn** [comment]
> Was the cluster started with consistent-topology-changes feature?

@enaydanov as @nuivall suggested, can we restart the test with consistent-topology-changes feature please?

**Comment by enaydanov** [comment]
I started the test with consistent-topology-changes enabled: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/evgeniy/job/longevity-1tb-5days-azure-test/3/

btw, next build of the original job passed: https://jenkins.scylladb.com/job/scylla-master/job/tier1/job/longevity-1tb-5days-azure-test/5/

**Comment by k0machi** [comment]
Reproduced recently on `#15` of the same job

## Packages

Scylla version: `5.5.0~dev-20240506.8472c46c8ab9` with build-id `04a94b62e8db49e0d71c8a4b56f6e7cf6574933e`

Kernel Version: `5.15.0-1061-azure`


## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

*Describe your issue in detail and steps it took to produce it.*

## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Cluster size: 4 nodes (Standard_L16s_v3)

Scylla Nodes used in this run:
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-7 (172.174.250.11 | 10.0.0.6) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-6 (52.146.0.134 | 10.0.0.14) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-5 (52.146.7.53 | 10.0.0.14) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-4 (172.178.89.221 | 10.0.0.8) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-3 (172.178.89.139 | 10.0.0.7) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-2 (172.178.88.247 | 10.0.0.6) (shards: 14)
- longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-1 (20.102.118.100 | 10.0.0.5) (shards: 14)


OS / Image: `/subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/SCYLLA-IMAGES/providers/Microsoft.Compute/images/scylla-5.5.0-dev-x86_64-2024-05-07T02-24-09` (azure: undefined_region)

Test: `longevity-1tb-5days-azure-test`
Test id: `7e11c3b8-57bf-4453-9760-ac12261a4d68`
Test name: `scylla-master/tier1/longevity-1tb-5days-azure-test`
Test config file(s):

- [longevity-1TB-5days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/3b8467a12f51992e6f212479fbf97b630ad67ea6/test-cases/longevity/longevity-1TB-5days-authorization-and-tls-ssl.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 7e11c3b8-57bf-4453-9760-ac12261a4d68`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=7e11c3b8-57bf-4453-9760-ac12261a4d68)
- Show all stored logs command: `$ hydra investigate show-logs 7e11c3b8-57bf-4453-9760-ac12261a4d68`


## Logs:
- **db-cluster-7e11c3b8.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/db-cluster-7e11c3b8.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/db-cluster-7e11c3b8.tar.gz)
- **sct-runner-events-7e11c3b8.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/sct-runner-events-7e11c3b8.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/sct-runner-events-7e11c3b8.tar.gz)
- **sct-7e11c3b8.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/sct-7e11c3b8.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/sct-7e11c3b8.log.tar.gz)
- **loader-set-7e11c3b8.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/loader-set-7e11c3b8.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/loader-set-7e11c3b8.tar.gz)
- **monitor-set-7e11c3b8.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/monitor-set-7e11c3b8.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7e11c3b8-57bf-4453-9760-ac12261a4d68/20240519_055017/monitor-set-7e11c3b8.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/tier1/job/longevity-1tb-5days-azure-test/15/)
[Argus](https://argus.scylladb.com/test/1e333df8-a7e8-4171-8ab7-1d7bdea907d5/runs?additionalRuns[]=7e11c3b8-57bf-4453-9760-ac12261a4d68)
</details>

**Comment by mykaul** [comment]
> > Was the cluster started with consistent-topology-changes feature?
> 
> @enaydanov as @nuivall suggested, can we restart the test with consistent-topology-changes feature please?

@ptrsmrn - can you re-look at this?

**Comment by ptrsmrn** [comment]
@k0machi @enaydanov I see that this job sometimes passes, sometimes fails (pic attached), so it's a flaky test, correct?
![image](https://github.com/scylladb/scylladb/assets/124208650/cb5c031a-3f7d-4825-951d-3cd89afbde7e)

**Comment by ptrsmrn** [comment]
And just so we're on the same page, can anybody please confirm it's this test that failed? https://github.com/scylladb/scylla-cluster-tests/blob/master/longevity_test.py#L116

**Comment by mykaul** [comment]
> @k0machi @enaydanov I see that this job sometimes passes, sometimes fails (pic attached), so it's a flaky test, correct? ![image](https://private-user-images.githubusercontent.com/124208650/337212331-cb5c031a-3f7d-4825-951d-3cd89afbde7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2Njc5MjIsIm5iZiI6MTcxNzY2NzYyMiwicGF0aCI6Ii8xMjQyMDg2NTAvMzM3MjEyMzMxLWNiNWMwMzFhLTNmN2QtNDgyNS05NTFkLTNjZDg5YWZiZGU3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNlQwOTUzNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ODQ2MmUwYWEyMmZlMDg4MGEwN2Y1NjAxNDFlZGNhZTdhODY5NjQzNWIyZjgwMTZiYTVmZWMxMmQzYjVhZTEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.HdHR6KzRB5Dm2f9Trc9KqnmTRrlyHgBx5jNWsYDptig)

No, it's unlikely to be flaky. It is more likely that sometimes it stumbles on Scylla bugs, and sometimes it doesn't. Unlike dtest/unit tests, I think we see this pattern more often with longevity tests.

**Comment by nuivall** [comment]
with run 15 (https://argus.scylladb.com/test/1e333df8-a7e8-4171-8ab7-1d7bdea907d5/runs?additionalRuns%5B%5D=a6095eee-a478-42f1-8d4c-5b35b06ac790) failure seems to be different than originally reported:

```
Stress command completed with bad status 1: WARN  05:38:08,787 Not using advanced port-based shard awareness with /10.0.0.5:9042 because we're m"
```

So as I understand we fail the test because cs returned 1, and it did due to some warning
- you probably shouldn't fail test on warning
- you need to fix error reporting because it's clipping the warning message

**Comment by mykaul** [comment]
@nuivall - I doubt it (but we should check) that we've failed on warning. Most likely, we failed on timeout , and the output was clipped and showed only the beginning - there should be a complete loader log somewhere.

**Comment by ptrsmrn** [comment]
Another thing is that we see in `sct-7e11c3b8.log` these logs:
```
< t:2024-05-19 05:38:10,177 f:base.py         l:152  c:RemoteLibSSH2CmdRunner p:DEBUG > WARN  05:35:54,469 Error creating netty channel to /10.0.0.8:9042
< t:2024-05-19 05:38:10,177 f:base.py         l:152  c:RemoteLibSSH2CmdRunner p:DEBUG > com.datastax.shaded.netty.channel.AbstractChannel$AnnotatedNoRouteToHostException: No route to host: /10.0.0.8:9042
```
And logs from `longevity-tls-1tb-7d-master-db-node-7e11c3b8-eastus-4/messages.log` say that the node died (abruptly?) and wasn't started, hence the above "no route to host logs":
```
...
2024-05-18T03:26:01.311+00:00 longevity-tls-1tb-7d-master-db-node-eastus-4     !INFO | scylla[7215]:  [shard  0:main] init - seeds={10.0.0.5}, listen_address=10.0.0.8, broadcast_address=10.0.0.8
...
// the last log lines are:
2024-05-18T22:56:39.352+00:00 longevity-tls-1tb-7d-master-db-node-eastus-4     !INFO | scylla[860]:  [shard  6:comp] compaction - [Compact keyspace1.standard1 e2954f00-1569-11ef-85f1-a0a95683d4ff] Compacting [/var/lib/scylla/data/keyspace1/standard1-c7dbc15014c711ef84d60ef8c6fc0e25/me-3gg8_1qig_4c90021bu883kklfcv-big-Da...
2024-05-18T22:56:40.375+00:00 longevity-tls-1tb-7d-master-db-node-eastus-4   !NOTICE | syslog-ng[653]: syslog-ng shutting down; version='3.35.1'
```
Note the dates and times, node 4 died 6h+ before c-s wants to connect to it.

**Comment by mykaul** [comment]
@ptrsmrn - it'd be hard to do so just from one side, but it may well be that the node was killed as part of a Nemesis. And yes, nodes come and go and c-s may lost connections to them. It's not out of the ordinary.

**Comment by nuivall** [comment]
Indeed error reporting is misleading here, in the loader file there is a mix of stderr and stdout
```
java.lang.AssertionError
read1,      56713916,     345,     345,     345,     1.5,     1.9,     2.6,     2.6,     2.6,     2.6,86400.1,  0.00089,      0,      0,       0,       0,       0,       0
	at com.datastax.driver.core.ConvictionPolicy$DefaultConvictionPolicy.signalConnectionClosed(ConvictionPolicy.java:90)
read2,      56698966,     968,     968,     968,     1.8,     1.8,     3.5,     5.6,     5.6,     5.6,86400.1,  0.00089,      0,      0,       0,       0,       0,       0
	at com.datastax.driver.core.Connection.closeAsync(Connection.java:1095)
read3,      56686267,     823,     823,     823,     2.0,     1.8,     4.6,     6.5,     6.5,     6.5,86400.1,  0.00089,      0,      0,       0,       0,       0,       0
	at com.datastax.driver.core.HostConnectionPool.discardAvailableConnections(HostConnectionPool.java:1011)
total,     340109102,    4827,    4827,    4827,     2.1,     2.0,     3.9,     4.6,     6.5,     6.5,86400.1,  0.00089,      0,      0,       0,       0,       0,       0
	at com.datastax.driver.core.HostConnectionPool.closeAsync(HostCo
```

which translates to failing assert:
```
    void signalConnectionClosed(Connection connection) {
      int remaining = openConnections.decrementAndGet();
      assert remaining >= 0;
```

looks like a bug in CS still, why it closed more connections than it opened? It's very hard to get to scylla part while the CS error reporting looks like this.

**Comment by nuivall** [comment]
I still think that error reporting in longevity tests should be improved because otherwise how we can de-duplicate failures?

**Comment by ptrsmrn** [comment]
@k0machi @enaydanov WRT https://github.com/scylladb/scylladb/issues/17782#issuecomment-2151921089 can you please help us understand why this node would be down? Is it that c-s killed it and "forgot" to start maybe? `sct-7e11c3b8.log` has ~1M lines, it's hard to extract info out of it.

**Comment by k0machi** [comment]
> @k0machi @enaydanov WRT [#17782 (comment)](https://github.com/scylladb/scylladb/issues/17782#issuecomment-2151921089) can you please help us understand why this node would be down? Is it that c-s killed it and "forgot" to start maybe? `sct-7e11c3b8.log` has ~1M lines, it's hard to extract info out of it.

@ptrsmrn It was decommissioned earlier in the test:

Run from @enaydanov:

![chrome_cMq18zOQqV](https://github.com/scylladb/scylladb/assets/7761415/3c2d95d0-3aca-4ca8-907a-f1956b062ae3)
![chrome_ABVTGnJw07](https://github.com/scylladb/scylladb/assets/7761415/a7d48b25-aa97-49cd-bde5-fd82ae9aa493)

Mine:

![image](https://github.com/scylladb/scylladb/assets/7761415/671e93c8-c1c1-4957-b7b9-7da20f83e4b5)
![chrome_beYR7DJltr](https://github.com/scylladb/scylladb/assets/7761415/cbd87270-42e5-451e-b23e-0162e8d0dd7b)

**Comment by ptrsmrn** [comment]
I and @k0machi met but couldn't identify the root cause. I'm not sure whether it's a scylla problem or a c-s problem. Those 2 runs failed on the same step `disrupt_destroy_data_then_rebuild`, but logs from those 2 runs are different.
In `sct-a6095eee.log` from first run we see: 
```
< t:2024-03-09 23:02:59,034 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG > com.datastax.driver.core.exceptions.ReadTimeoutException: Cassandra timeout during read query at consistency QUORUM (2 responses were required but only 1 replica responded). In case this was generated during read repair, the consistency level is not representative of the actual consistency.
...
< t:2024-03-10 03:21:01,603 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG > com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 1 replica responded, 1 failed)
..
< t:2024-03-10 03:21:01,603 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG > com.datastax.driver.core.exceptions.ReadFailureException: Cassandra failure during read query at consistency QUORUM (3 responses were required but only 0 replica responded, 1 failed)
```
but these messages are not present in the 2nd run in `sct-7e11c3b8.log`. Anyways, just "lost quorum" messages are not enough to indicate any scylla's issue.

**Comment by juliayakovlev** [comment]
Looks like the issue was reproduced again (I am not sure 100%, that it is)

Mix c-s load failed with:
```
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt; Caused by: java.net.ConnectException: Connection refused
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java
:334)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at java.base/java.lang.Thread.run(Thread.java:829)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt; java.lang.AssertionError
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.ConvictionPolicy$DefaultConvictionPolicy.signalConnectionClosed(ConvictionPolicy.java:9
0)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.Connection.closeAsync(Connection.java:1095)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.HostConnectionPool.discardAvailableConnections(HostConnectionPool.java:1011)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.HostConnectionPool.closeAsync(HostConnectionPool.java:972)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.SessionManager.closeAsync(SessionManager.java:196)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.Cluster$Manager.close(Cluster.java:1989)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.Cluster$Manager.access$200(Cluster.java:1560)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.Cluster.closeAsync(Cluster.java:625)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.driver.core.Cluster.close(Cluster.java:636)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at org.apache.cassandra.stress.util.JavaDriverClient.disconnect(JavaDriverClient.java:286)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at org.apache.cassandra.stress.settings.StressSettings.disconnect(StressSettings.java:412)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at org.apache.cassandra.stress.StressAction.run(StressAction.java:98)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at org.apache.cassandra.stress.Stress.run(Stress.java:143)
&lt; t:2024-06-10 12:03:19,781 f:base.py         l:150  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at org.apache.cassandra.stress.Stress.main(Stress.java:62)
&lt; t:2024-06-10 12:03:19,840 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  &gt; 2024-06-10 12:03:19.818: (CassandraStressEvent Severity.CRITICAL) period_type=end event_id=e3103e35-882a-4bb4-8bad-7580652d65e9 during_nemesis=RestartWithResharding duration=12h2m31s: node=Node longevity-tls-1tb-7d-2023-1-loader-node-2b3bf1c3-eastus-2 [172.212.85.89 | 10.0.0.10]
&lt; t:2024-06-10 12:03:19,840 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  &gt; stress_cmd=cassandra-stress mixed cl=QUORUM duration=720m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=100 -pop seq=550100151..1100200300  -log interval=5 -col 'size=FIXED(200) n=FIXED(5)'
&lt; t:2024-06-10 12:03:19,840 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  &gt; errors:
&lt; t:2024-06-10 12:03:19,840 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  &gt; 
&lt; t:2024-06-10 12:03:19,840 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  &gt; Stress command completed with bad status 1: WARN  00:00:51,941 Not using advanced port-based shard awareness with /10.0.0.8:9042 because we're m
```

15 minutes before load failure `Pool is INITIALIZING` exceptions, complains about `10.0.0.6` (node6).
In time of `Pool is INITIALIZING` messages the `node6` was running and OK, not under any nemesis.
Before starting to receive `Pool is INITIALIZING messages` the raft group was re-registered on the node6:
```
&lt; t:2024-06-10 11:43:17,216 f:db_log_reader.py l:123  c:sdcm.db_log_reader   p:DEBUG &gt; 2024-06-10T11:43:14.843+00:00 longevity-tls-1tb-7d-2023-1-db-node-eastus-6     !INFO | scylla[10554]:  [shard  0] raft_group_registry - marking Raft server b37d9603-1b17-4032-9eda-7c9eb7b2af0a as dead for raft groups
&lt; t:2024-06-10 11:43:17,258 f:db_log_reader.py l:123  c:sdcm.db_log_reader   p:DEBUG &gt; 2024-06-10T11:43:15.097+00:00 longevity-tls-1tb-7d-2023-1-db-node-eastus-6     !INFO | scylla[10554]:  [shard  0] raft_group_registry - marking Raft server b37d9603-1b17-4032-9eda-7c9eb7b2af0a as alive for raft groups
&lt; t:2024-06-10 11:43:17,519 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG &gt;         at com.datastax.shaded.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt; Reconnection error details &lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt; Reconnection error details
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt; Traceback (most recent call last):
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt;   File "cassandra/pool.py", line 287, in cassandra.pool._ReconnectionHandler.run
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt;   File "cassandra/pool.py", line 358, in cassandra.pool._HostReconnectionHandler.try_reconnect
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt;   File "cassandra/connection.py", line 875, in cassandra.connection.Connection.factory
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt;   File "/usr/local/lib/python3.10/site-packages/cassandra/io/libevreactor.py", line 267, in __init__
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt;     self._connect_socket()
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt;   File "cassandra/connection.py", line 954, in cassandra.connection.Connection._connect_socket
&lt; t:2024-06-10 11:43:17,525 f:thread.py       l:58   c:cassandra.pool       p:DEBUG &gt; ConnectionRefusedError: [Errno 111] Tried connecting to [('10.0.0.7', 9042)]. Last error: Connection refused

&lt; t:2024-06-10 11:43:47,026 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG &gt; WARN  11:43:08,767 Unexpected error while querying /10.0.0.6:9042 - [com.datastax.driver.core.exceptions.ConnectionException: [/10.0.0.6:9042] Pool is INITIALIZING]. Find next host to query.
&lt; t:2024-06-10 11:43:47,529 f:base.py         l:228  c:RemoteLibSSH2CmdRunner p:DEBUG &gt; WARN  11:43:08,767 Unexpected error while querying /10.0.0.6:9042 - [com.datastax.driver.core.exceptions.ConnectionException: [/10.0.0.6:9042] Pool is INITIALIZING]. Find next host to query.
```


## Packages

Scylla version: `2023.1.9-20240609.4a93c32572b9` with build-id `4aea949e59e20db48d10bc09ed9af75aa98dfd9a`





Kernel Version: `5.15.0-1064-azure`


## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

*Describe your issue in detail and steps it took to produce it.*

## Impact

*Describe the impact this issue causes to the user.*

## How frequently does it reproduce?

*Describe the frequency with how this issue can be reproduced.*

## Installation details

Cluster size: 4 nodes (Standard_L16s_v3)

Scylla Nodes used in this run:
- longevity-tls-1tb-7d-2023-1-db-node-2b3bf1c3-eastus-6 (48.217.64.26 | 10.0.0.6) (shards: 14)
- longevity-tls-1tb-7d-2023-1-db-node-2b3bf1c3-eastus-5 (172.210.76.183 | 10.0.0.14) (shards: 14)
- longevity-tls-1tb-7d-2023-1-db-node-2b3bf1c3-eastus-4 (172.212.84.18 | 10.0.0.8) (shards: 14)
- longevity-tls-1tb-7d-2023-1-db-node-2b3bf1c3-eastus-3 (172.212.85.31 | 10.0.0.7) (shards: 14)
- longevity-tls-1tb-7d-2023-1-db-node-2b3bf1c3-eastus-2 (172.212.85.30 | 10.0.0.6) (shards: 14)
- longevity-tls-1tb-7d-2023-1-db-node-2b3bf1c3-eastus-1 (172.212.85.15 | 10.0.0.5) (shards: 14)


OS / Image: `/subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/scylla-images/providers/Microsoft.Compute/images/scylla-2023.1.9-x86_64-2024-06-09T22-50-54` (azure: undefined_region)

Test: `longevity-1tb-5days-azure-test`
Test id: `2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1`
Test name: `enterprise-2023.1/longevity/longevity-1tb-5days-azure-test`
Test config file(s):

- [longevity-1TB-5days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/884b6de8a312e183df7b984aa3de681c9b1a6a21/test-cases/longevity/longevity-1TB-5days-authorization-and-tls-ssl.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1)
- Show all stored logs command: `$ hydra investigate show-logs 2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1`


## Logs:
- **db-cluster-2b3bf1c3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/db-cluster-2b3bf1c3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/db-cluster-2b3bf1c3.tar.gz)
- **sct-runner-events-2b3bf1c3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/sct-runner-events-2b3bf1c3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/sct-runner-events-2b3bf1c3.tar.gz)
- **sct-2b3bf1c3.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/sct-2b3bf1c3.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/sct-2b3bf1c3.log.tar.gz)
- **loader-set-2b3bf1c3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/loader-set-2b3bf1c3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/loader-set-2b3bf1c3.tar.gz)
- **monitor-set-2b3bf1c3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/monitor-set-2b3bf1c3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1/20240610_120652/monitor-set-2b3bf1c3.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2023.1/job/longevity/job/longevity-1tb-5days-azure-test/16/)
[Argus](https://argus.scylladb.com/test/bf1550bc-0943-4e5a-ab22-03fcc781d580/runs?additionalRuns[]=2b3bf1c3-d55d-488f-8dfd-6193dfe5c9e1)
</details>

**Comment by ptrsmrn** [comment]
The test was run with auth v2 (consistent topology enabled). Per @nuivall https://github.com/scylladb/scylladb/issues/17782#issuecomment-2151960507 it does look like a java driver issue, because the test runs exactly for 24:00 hours and it has problems cleaning up opened connections - seems like the driver lost track of how many connection are opened, which triggers an assertion that not all connections have been closed when disconnecting from a cluster, which fails the test. 
@avelanarius please take a look.

**Comment by mykaul** [comment]
Lost momentum here. Please re-open if reproduced.
